src/kubernetes.cc

```diff
@@ -1361,8 +1342,10 @@ void KubernetesUpdater::NotifyStopUpdater() {
 }
 
 void KubernetesUpdater::MetadataCallback(
+    const std::string& watch_name, const Timestamp& collected_at,
```

Comment:
Why wouldn't you extract this data from `result`? Better yet, why not do this in `UpdateMetadataCallback`?